marconi-chain-index/src/Marconi/ChainIndex/Experimental/Indexers/EpochState.hs

```diff
@@ -0,0 +1,595 @@
+  -> FilePath
+  -> m (Core.FileIndexer EpochMetadata EpochState)
+buildEpochStateIndexer codecConfig path =
+  let serialiseLedgerState =
+        CBOR.toLazyByteString
+          . O.encodeExtLedgerState
+            (O.encodeDisk codecConfig)
+            (O.encodeDisk codecConfig)
+            (O.encodeDisk codecConfig)
+          . extLedgerState
+
+      deserialiseLedgerState (EpochMetadata Nothing _) = const (Right Nothing)
+      deserialiseLedgerState (EpochMetadata (Just blockNo') _) =
+        bimap
+          (Text.pack . show)
+          (Just . flip EpochState blockNo' . snd)
+          . CBOR.deserialiseFromBytes
+            ( O.decodeExtLedgerState
+                (O.decodeDisk codecConfig)
+                (O.decodeDisk codecConfig)
+                (O.decodeDisk codecConfig)
+            )
+      blockNoAsText = maybe "" (Text.pack . show . (\(C.BlockNo b) -> b) . blockNo)
+      metadataAsText (Core.Timed C.ChainPointAtGenesis evt) = [blockNoAsText evt]
+      metadataAsText (Core.Timed chainPoint evt) =
+        let chainPointTexts = case chainPoint of
+              C.ChainPoint (C.SlotNo slotNo) blockHeaderHash ->
+                [Text.pack $ show slotNo, C.serialiseToRawBytesHexText blockHeaderHash]
+         in blockNoAsText evt : chainPointTexts
+   in Core.mkFileIndexer
+        path
+        False
+        "epochState"
+        "cbor"
+        serialiseLedgerState
+        deserialiseLedgerState
+        metadataAsText
+        deserialiseMetadata
+        metadataChainpoint
+
+buildBlockIndexer
+  :: (MonadIO m, MonadError Core.IndexerError m)
+  => O.CodecConfig (O.HardForkBlock (O.CardanoEras O.StandardCrypto))
+  -> FilePath
+  -> m (Core.FileIndexer EpochMetadata (C.BlockInMode C.CardanoMode))
+buildBlockIndexer codecConfig path =
+  let serialiseBlock =
+        CBOR.toLazyByteString . O.encodeDisk codecConfig . C.toConsensusBlock
+      deserialiseBlock (EpochMetadata Nothing _) = const (Right Nothing)
+      deserialiseBlock _ =
+        bimap
+          (Text.pack . show)
+          (Just . C.fromConsensusBlock C.CardanoMode)
+          . fmap (\(bs', decode) -> decode bs')
+          . CBOR.deserialiseFromBytes (O.decodeDisk codecConfig)
+      blockNoAsText = maybe "" (Text.pack . show . (\(C.BlockNo b) -> b) . getBlockNo)
+      metadataAsText (Core.Timed C.ChainPointAtGenesis evt) = [blockNoAsText evt]
+      metadataAsText (Core.Timed chainPoint evt) =
+        let chainPointTexts = case chainPoint of
+              C.ChainPoint (C.SlotNo slotNo) blockHeaderHash ->
+                [Text.pack $ show slotNo, C.serialiseToRawBytesHexText blockHeaderHash]
+         in blockNoAsText evt : chainPointTexts
+   in Core.mkFileIndexer
+        path
+        True
+        "block"
+        "cbor"
+        serialiseBlock
+        deserialiseBlock
+        metadataAsText
+        deserialiseMetadata
+        metadataChainpoint
+
+-- TODO placeholder, to implement
```

Comment:
Bin?

Reply:
no, we definitely need to add logic to them, I'll create the ticket (at the moment we just store points)